src/stdlib/inspector.ts

```diff
@@ -0,0 +1,44 @@
+import { Node } from 'estree'
+import { Context, Result } from '..'
+import { Scheduler, Value } from '../types'
+
+export const saveState = (
+  context: Context,
+  it: IterableIterator<Value>,
+  scheduler: Scheduler
+): void => {
+  context.debugger.state.it = it
+  context.debugger.state.scheduler = scheduler
+}
+
+export const setBreakpointAtLine = (lines: string[]): void => {
+  breakpoints = lines
+}
+
+export const manualToggleDebugger = (context: Context): Result => {
+  context.runtime.break = true
+  return {
+    status: 'suspended',
+    scheduler: context.debugger.state.scheduler,
+    it: context.debugger.state.it,
+    context
+  }
+}
+
+let breakpoints: string[] = []
+let moved: boolean = true
+let prevStoppedLine: number = -1
+
+export const checkEditorBreakpoints = (context: Context, node: Node): void => {
+  if (node.loc) {
+    const currentLine = node.loc.start.line - 1
+    if (!moved && currentLine !== prevStoppedLine) {
+      moved = true
+    }
+    if (context.runtime.debuggerOn && breakpoints[currentLine] === 'ace_breakpoint' && moved) {
```

Comment:
Haven't played with it too much yet to review, but we're trying to have js-slang be independent of cadet-frontend, so I'm not sure about this 'ace_breakpoint'.

Reply:
This is just something carried over from the editor, we could change it to be as long as it is not undefined.